### DISCUSSION TITLE
Global encrypt doesn't take effect

### DISCUSSION THREAD

**Discussion by yzqdev** [vuepress-theme-hope/vuepress-theme-hope#1754] [Q&A] [answered]
Accoding to [https://vuepress-theme-hope.github.io/v2/guide/feature/encrypt.html#global-encryption](https://vuepress-theme-hope.github.io/v2/guide/feature/encrypt.html#global-encryption)
I have only set the following properties.
```
  encrypt: {
    global: true,
    admin: ['1234', '123456'],
  },
```
but it doesn't seem to take effect.

Here is the mini repo . [stackblitz ](https://stackblitz.com/edit/vuepress-theme-hope-svqrm7?file=docs%2F.vuepress%2FthemeConfig.ts)

What is  expected is like this, we could see nothing before we enter the password.
![image](https://user-images.githubusercontent.com/20037151/166132562-ba68a6ed-aec0-45cc-b75b-b0033c438205.png)

is my configuration wrong?

**Comment by Mister-Hope** [comment]
Appreciate for providing detailed reproduction, I will have a look later

**Comment by Mister-Hope** [comment]
Your configuration is invalid, as you are providing mutiple `encrypt` key in your config, and only the last one has effect.

**Reply by yzqdev** [reply]
Now I have deleted the duplicate `encrypt` key, it seemed to be the same.
do i need to also set `themeConfig.encrypt.confg` property when i use encrypt global?

https://stackblitz.com/edit/vuepress-theme-hope-svqrm7?file=docs%2F.vuepress%2FthemeConfig.ts

**Reply by Mister-Hope** [reply]
I will have a look later, after merging #1731

**Reply by yzqdev** [reply]
Great, thank you.

**Comment by Mister-Hope** [comment, marked as answer]
Fixed in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/1401242ed87ce6bc7f995b5ba0defbab27cb0d2d

**Reply by yzqdev** [reply]
Got it, thank you !

**Comment by yzqdev** [comment]
It seems the problem still exists.

site https://yzqbugs.github.io/hope-demo/
and it's minimal repo https://github.com/yzqbugs/hope-demo  
I'm using `vuepress 2.0.0-beta.43` and `vuepress-theme-hope 2.0.0-beta.52`

I've set those properties
```
 encrypt: {
    global: true,
    admin: "123456",
  },
```
Maybe the correct page looks like this ?
![img](https://user-images.githubusercontent.com/20037151/166132562-ba68a6ed-aec0-45cc-b75b-b0033c438205.png)

**Reply by Mister-Hope** [reply]
Fixed in latest version.

**Reply by Mister-Hope** [reply]
Besides, we will add remember password support in future versions.

**Reply by yzqdev** [reply]
Awesome! Thank you😃
